TriblerGUI/tribler_window.py

```diff
@@ -532,6 +532,12 @@ def received_settings(self, settings):
         if 'error' in settings:
             raise RuntimeError(TriblerRequestManager.get_message_from_error(settings))
 
+        # If there is any pending dialog (likely download dialog or error dialog of setting not available),
+        # close the dialog
+        if self.dialog:
```

Comment:
I don't get it, why do we need this? In what kind of situation this should trigger? This can become a source of race conditions in the GUI and make its behavior non-deterministic.

Reply:
I was able to get to this situation trying to reproduce nubro's issue. This happens if you're opening a torrent file with Tribler by clicking it. Tribler will try to load the download dialog but if the setting is not fetched it, it will show an error dialog and the UI gets stuck on it. Note the changes in made in `start_download_from_uri()`. By re-adding the download uri to the pending list, the request is re-processed when the settings is received. With the above code, it makes sure that error dialog is closed and proper download dialog is shown.

Reply:
Excellent!
Please, put this:

>By re-adding the download uri to the pending list, the request is re-processed when the settings is received.

in a comment in the code.

It was not obvious for me during the review, so it should be commented so we do not forget why we need this :wink: